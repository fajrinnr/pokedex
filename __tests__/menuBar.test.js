import React from "react";
import renderer from "react-test-renderer";
import MenuBar from "../src/components/menuBar";
import Enzyme, { mount } from "enzyme";
import Router from "next/router";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

// jest.mock("next/link", () => {
//   return ({ children }) => {
//     return children;
//   };
// });

// describe("Test of <HeaderContainer>", () => {
//   test("Test the page redirect after click", async (done) => {
//     const wrapper = mount(<MenuBar />);
//     await wrapper.find("#pokemon-list-link").at(0).simulate("click");
//     // expect the page getting redirect
//   });
// });

describe("Test of <HeaderContainer>", () => {
  const spies = {};

  beforeEach(() => {
    spies.routerChangeStart = jest.fn();
    Router.events.on("routeChangeStart", spies.routerChangeStart);
  });

  afterEach(() => {
    Router.events.off("routeChangeStart", spies.routerChangeStart);
  });

  test("Test the page redirect after click", async (done) => {
    const wrapper = mount(<MenuBar />);
    await wrapper.find("#pokemon-list-link").at(0).simulate("click");

    expect(spies.routerChangeStart).toHaveBeenCalledWith("/");
  });
});
