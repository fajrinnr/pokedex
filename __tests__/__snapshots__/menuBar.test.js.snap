// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
<div
  className="css-4v8ryg-MenuBar"
  data-testid="container-link"
>
  <div
    className="css-1vfxa0x-MenuBar"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      alt="Pikachu"
      className="css-1i0yynm-MenuBar"
      src="/pikachu.svg"
    />
    <p
      className="css-zotcb1-MenuBar"
    >
      Pokemon List
    </p>
  </div>
  <div
    className="css-1vfxa0x-MenuBar"
    data-testid="my-pokemon-link"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      alt="Poke Ball"
      className="css-1i0yynm-MenuBar"
      src="/pokeballs.svg"
    />
    <p
      className="css-zotcb1-MenuBar"
    >
      My Pokemon
    </p>
  </div>
</div>
`;
